Sources/Turf/Spline.swift

```diff
@@ -0,0 +1,147 @@
+import Foundation
+#if !os(Linux)
+import CoreLocation
+#endif
+
+
+struct SplinePoint {
+    let x: CLLocationDegrees
+    let y: CLLocationDegrees
+    let z: CLLocationDegrees
+    
+    init(coordinate: CLLocationCoordinate2D) {
+        x = coordinate.longitude
+        y = coordinate.latitude
+        z = 0
+    }
+    
+    init(x: CLLocationDegrees, y: CLLocationDegrees, z: CLLocationDegrees) {
+        self.x = x
+        self.y = y
+        self.z = z
+    }
+    
+    var coordinate: CLLocationCoordinate2D {
+        return CLLocationCoordinate2D(latitude: y, longitude: x)
+    }
+}
+
+struct Spline {
+    private let points: [SplinePoint]
+    private let duration: Int
+    private let sharpness: Double
+    private let stepLength: Double
+    private let length: Int
+    private let delay: Int = 0
+    private var centers = [SplinePoint]()
+    private var controls = [(SplinePoint, SplinePoint)]()
+    private var steps = [Int]()
+    
+    init?(points: [SplinePoint], duration: Int, sharpness: Double, stepLength: Double = 60) {
+        guard points.count >= 2 else { return nil }
+        self.points = points
+        self.duration = duration
+        self.sharpness = sharpness
+        self.stepLength = stepLength
+        
+        length = points.count
+        
+        for index in stride(from: 0, to: points.count - 1, by: 1) {
+            let point = points[index]
+            let nextPoint = points[index + 1]
+            let center = SplinePoint(x: (point.x + nextPoint.x) / 2, y: (point.y + nextPoint.y) / 2, z: (point.z + nextPoint.z) / 2)
+            centers.append(center)
+        }
+        
+        controls.append((points[0], points[0]))
+        
+        for index in stride(from: 0, to: centers.count - 1, by: 1) {
```

Comment:
Similarly, this is equivalent to a range. I probably wouldn’t use `zip(_:_:)` here because we’re juggling `nextPoint` in addition to `center` and `nextCenter`.